### PR TITLE
[Build] Use C++17 Constructor for tiktoken.cpp when C++20 is unavailable

### PR DESCRIPTION
**Pull request by python3kgae**
The basic_string_view constructor:
```
template< class It, class End >
constexpr basic_string_view( It first, End last );
```
requires C++20.

To allow the code to compile with C++17, use the basic_string_view constructor:
```
constexpr basic_string_view( const CharT* s, size_type count );
```
when C++20 is not available

For #4661

### CODE REVIEW

**Comment by cccclai** [comment]
Seems fine to me, will wait for CI signal

**Review by larryliu0820 — APPROVED** [review]

**Comment by larryliu0820** [comment]
Can you click the "Squash and merge" button with a good commit message? We are testing out landing PRs from GH.

**Comment by python3kgae** [comment]
> Can you click the "Squash and merge" button with a good commit message? We are testing out landing PRs from GH.

Don't have write access :(
Here's the commit message.

```
[Build] Use C++17 Constructor for basic_string_view in tiktoken.cpp when C++20 is unavailable

The basic_string_view constructor:

template< class It, class End >
constexpr basic_string_view( It first, End last );

requires C++20.

To allow the code to compile with C++17, use the basic_string_view constructor:

constexpr basic_string_view( const CharT* s, size_type count );

when C++20 is not available

For #4661
```

**Comment by kirklandsign** [comment]
> > Can you click the "Squash and merge" button with a good commit message? We are testing out landing PRs from GH.
> 
> Don't have write access :( Here's the commit message.
> 
> ```
> [Build] Use C++17 Constructor for basic_string_view in tiktoken.cpp when C++20 is unavailable
> 
> The basic_string_view constructor:
> 
> template< class It, class End >
> constexpr basic_string_view( It first, End last );
> 
> requires C++20.
> 
> To allow the code to compile with C++17, use the basic_string_view constructor:
> 
> constexpr basic_string_view( const CharT* s, size_type count );
> 
> when C++20 is not available
> 
> For #4661
> ```

Thank you for the feedback! We will update the rules on our side and let you know

**Comment by kirklandsign** [comment]
Seems that currently we need reviewers to merge the commit. I will do it.

**Pull request merged**
